File: app/src/main/java/activity/amigosecreto/util/SnackbarHelper.java
package activity.amigosecreto.util;

import android.view.View;
import androidx.core.content.ContextCompat;
import com.google.android.material.snackbar.Snackbar;
import activity.amigosecreto.R;

/**
 * Helper class for creating consistent Snackbar messages
 */
public class SnackbarHelper {

    /**
     * Show success message
     */
    public static void showSuccess(View view, String message) {
        Snackbar snackbar = Snackbar.make(view, message, Snackbar.LENGTH_SHORT);
        snackbar.setBackgroundTint(ContextCompat.getColor(view.getContext(), R.color.colorSecondary));
        snackbar.setTextColor(ContextCompat.getColor(view.getContext(), R.color.white));
        snackbar.show();
    }

    /**
     * Show error message
     */
    public static void showError(View view, String message) {
        Snackbar snackbar = Snackbar.make(view, message, Snackbar.LENGTH_LONG);
        snackbar.setBackgroundTint(ContextCompat.getColor(view.getContext(), R.color.colorError));
        snackbar.setTextColor(ContextCompat.getColor(view.getContext(), R.color.white));
        snackbar.show();
    }

    /**
     * Show info message
     */
    public static void showInfo(View view, String message) {
        Snackbar snackbar = Snackbar.make(view, message, Snackbar.LENGTH_SHORT);
        snackbar.setBackgroundTint(ContextCompat.getColor(view.getContext(), R.color.colorPrimary));
        snackbar.setTextColor(ContextCompat.getColor(view.getContext(), R.color.white));
        snackbar.show();
    }

    /**
     * Show message with action
     */
    public static void showWithAction(View view, String message, String actionText, View.OnClickListener action) {
        Snackbar snackbar = Snackbar.make(view, message, Snackbar.LENGTH_LONG);
        snackbar.setAction(actionText, action);
        snackbar.setActionTextColor(ContextCompat.getColor(view.getContext(), R.color.colorAccent));
        snackbar.show();
    }
}

File: app/src/main/java/activity/amigosecreto/util/SnackbarHelper.kt
package activity.amigosecreto.util

import android.view.View
import androidx.core.content.ContextCompat
import com.google.android.material.snackbar.Snackbar
import activity.amigosecreto.R

/**
 * Helper class for creating consistent Snackbar messages
 */
object SnackbarHelper {

    @JvmStatic
    fun showSuccess(view: View, message: String) {
        Snackbar.make(view, message, Snackbar.LENGTH_SHORT).apply {
            setBackgroundTint(ContextCompat.getColor(view.context, R.color.colorSecondary))
            setTextColor(ContextCompat.getColor(view.context, R.color.white))
            show()
        }
    }

    @JvmStatic
    fun showError(view: View, message: String) {
        Snackbar.make(view, message, Snackbar.LENGTH_LONG).apply {
            setBackgroundTint(ContextCompat.getColor(view.context, R.color.colorError))
            setTextColor(ContextCompat.getColor(view.context, R.color.white))
            show()
        }
    }

    @JvmStatic
    fun showInfo(view: View, message: String) {
        Snackbar.make(view, message, Snackbar.LENGTH_SHORT).apply {
            setBackgroundTint(ContextCompat.getColor(view.context, R.color.colorPrimary))
            setTextColor(ContextCompat.getColor(view.context, R.color.white))
            show()
        }
    }

    @JvmStatic
    fun showWithAction(view: View, message: String, actionText: String, action: View.OnClickListener) {
        Snackbar.make(view, message, Snackbar.LENGTH_LONG).apply {
            setAction(actionText, action)
            setActionTextColor(ContextCompat.getColor(view.context, R.color.colorAccent))
            show()
        }
    }
}
